insta/src/App.js

```diff
@@ -1,34 +1,88 @@
 import React from "react";
 import "./index.css";
-import 'bootstrap/dist/css/bootstrap.min.css';
+import "bootstrap/dist/css/bootstrap.min.css";
 import dummyData from "./dummy-data.js"; //? thought we needed {} to import guess it has something to do with export default at the bottom
 import SearchBar from "./components/SearchBar/SearchBar";
 import PostContainer from "./components/PostContainer/PostContainer";
 
 class App extends React.Component {
   state = {
-    data: dummyData
+    data: [],
+    filtered: []
   };
 
-  // addComment = (comment) => {
-  //   const newComment = {
-  //     text: comment,
-  //     username: "",
-  //     id: Date.now()
-  //   }
-  //   this.setState(prevState => {
-  //     return {
-  //       data: prevState.data.map(post => {
+  //         Initialization             ///
+  componentDidMount() {
+    this.setState({
+      data: dummyData,
+      filtered: dummyData
+    });
+  }
+
+  //          ADD A COMMENT TO A POST            //
+  addComment = (id, input) => {
+    this.setState(prevState => {
+      return {
+        data: prevState.data.map(post => {
+          if (post.id === id) {
+            return {
+              ...post,
+              comments: [...post.comments, input]
+            };
+          } else {
+            return post;
+          }
+        })
+      };
+    });
+  };
+  //             LIKE A POST                     //
+  likePost = id => {
+    console.log(id);
+    this.setState(prevState => {
+      return {
+        data: prevState.data.map(post => {
+          if (post.id === id) {
+            return {
+              ...post,
+              likes: (post.likes += 1),
+              liked: true
+            };
```

Comment:
Super impressed on what you did for your addComment and likePost function, one thing I would do though to make your life easier is handling all the this functionality in a post component rather than a component that handles all the mapping out of data. 

For example, Ben's [`PostContainer`](https://github.com/Zurglots/React-Insta-Clone/blob/ben-feole/instaclone/src/components/PostContainer/PostContainer.js) is in charge of just mapping out each post and rendering a [`Post`](https://github.com/Zurglots/React-Insta-Clone/blob/ben-feole/instaclone/src/components/PostContainer/Post.js) component which is in charge of all the functionality for one post. 

Now for the comment part, Ben has rendered his [`CommentSection`](https://github.com/Zurglots/React-Insta-Clone/blob/ben-feole/instaclone/src/components/CommentSection/CommentSection.js) in his [`Post` (https://github.com/Zurglots/React-Insta-Clone/blob/ben-feole/instaclone/src/components/PostContainer/Post.js) component and then you can see how to kind of handle the addComment functionality in an easier manner.


but, again this is a preference thing what I suggested above was just a suggestion that can make your life easier! 






(Let me know if you saw this by replying with a comment 👍 )

Reply:
Little confused, are you just saying that the add a comment and like function should be in the individual post component instead of the main app component?